packages/devtools_app/lib/src/flutter/extent_delegate_list.dart

```diff
@@ -0,0 +1,464 @@
+    renderObject.extentDelegate = extentDelegate;
+  }
+}
+
+/// A sliver that contains multiple box children that each have known extent
+/// along the main axis.
+///
+/// This class is inspired by [RenderSliverFixedExtentBoxAdaptor] which provides
+/// similar functionality for the case where all items have the same extent.
+class RenderSliverExtentDelegateBoxAdaptor extends RenderSliverMultiBoxAdaptor {
+  /// Creates a sliver that contains multiple box children each of which have
+  /// extent along the main axis provided by [extentDelegate].
+  ///
+  /// The [childManager] argument must not be null.
+  RenderSliverExtentDelegateBoxAdaptor({
+    @required RenderSliverBoxChildManager childManager,
+    @required ExtentDelegate extentDelegate,
+  }) : super(childManager: childManager) {
+    this.extentDelegate = extentDelegate;
+  }
+
+  set extentDelegate(ExtentDelegate delegate) {
+    // We need to listen for when the delegate changes its layout.
+    delegate.onLayoutDirty = markNeedsLayout;
+    _extentDelegate = delegate;
+  }
+
+  ExtentDelegate _extentDelegate;
+
+  /// Called to estimate the total scrollable extents of this object.
+  ///
+  /// Must return the total distance from the start of the child with the
+  /// earliest possible index to the end of the child with the last possible
+  /// index.
+  ///
+  /// By default, defers to [RenderSliverBoxChildManager.estimateMaxScrollOffset].
+  @protected
+  double estimateMaxScrollOffset(
+    SliverConstraints constraints, {
+    int firstIndex,
+    int lastIndex,
+    double leadingScrollOffset,
+    double trailingScrollOffset,
+  }) {
+    return childManager.estimateMaxScrollOffset(
+      constraints,
+      firstIndex: firstIndex,
+      lastIndex: lastIndex,
+      leadingScrollOffset: leadingScrollOffset,
+      trailingScrollOffset: trailingScrollOffset,
+    );
+  }
+
+  int _calculateLeadingGarbage(int firstIndex) {
+    RenderBox walker = firstChild;
+    int leadingGarbage = 0;
+    while (walker != null && indexOf(walker) < firstIndex) {
+      leadingGarbage += 1;
+      walker = childAfter(walker);
+    }
+    return leadingGarbage;
+  }
+
+  int _calculateTrailingGarbage(int targetLastIndex) {
+    RenderBox walker = lastChild;
+    int trailingGarbage = 0;
+    while (walker != null && indexOf(walker) > targetLastIndex) {
+      trailingGarbage += 1;
+      walker = childBefore(walker);
+    }
+    return trailingGarbage;
+  }
+
+  BoxConstraints buildChildConstraints(int index) {
+    final currentItemExtent = _extentDelegate.itemExtent(index);
+    return constraints.asBoxConstraints(
+      minExtent: currentItemExtent,
+      maxExtent: currentItemExtent,
+    );
+  }
+
+  // This method is is a fork of RenderSliverFixedExtentBoxAdaptor.performLayout
+  // where we defer computations about offsets to the _extendDelegate and try
+  // to avoid logic only applicable if all children have the same extent.
+  @override
+  void performLayout() {
```

Comment:
can any of this method be put into helpers to make it more readable? Or are you trying to keep a similar structure to the flutter framework fork?

Reply:
yes this is the same structure as the code in the framework.